test/modules/image-sequencer.js

```diff
@@ -97,89 +97,92 @@ test('addSteps("name") adds a step', function (t) {
   t.equal(sequencer.images.test.steps[6].options.name, "channel", "Correct Step Added");
   t.end();
 });
 
 test('removeSteps("image",position) removes a step', function (t) {
-  sequencer.removeSteps('test',1);
+  sequencer.removeSteps('test', 1);
   t.equal(sequencer.images.test.steps.length, 6, "Length of steps reduced");
   t.end();
 });
 
 test('removeSteps({image: [positions]}) removes steps', function (t) {
-  sequencer.removeSteps('test',[1,2]);
+  sequencer.removeSteps('test', [1, 2]);
   t.equal(sequencer.images.test.steps.length, 4, "Length of steps reduced");
   t.end();
 });
 
 test('removeSteps(position) removes steps', function (t) {
-  sequencer.removeSteps([1,2]);
+  sequencer.removeSteps([1, 2]);
   t.equal(sequencer.images.test.steps.length, 2, "Length of steps reduced");
   t.end();
 });
 
 test('insertSteps("image",position,"module",options) inserts a step', function (t) {
-  sequencer.insertSteps('test',1,'channel',{});
+  sequencer.insertSteps('test', 1, 'channel', {});
   t.equal(sequencer.images.test.steps.length, 3, "Length of Steps increased");
   t.equal(sequencer.images.test.steps[1].options.name, "channel", "Correct Step Inserted");
   t.end();
 });
 
 test('insertSteps("image",position,"module") inserts a step', function (t) {
-  sequencer.insertSteps('test',1,'channel');
+  sequencer.insertSteps('test', 1, 'channel');
   t.equal(sequencer.images.test.steps.length, 4, "Length of Steps increased");
   t.equal(sequencer.images.test.steps[1].options.name, "channel", "Correct Step Inserted");
   t.end();
 });
 
 test('insertSteps(position,"module") inserts a step', function (t) {
-  sequencer.insertSteps(1,'channel');
+  sequencer.insertSteps(1, 'channel');
   t.equal(sequencer.images.test.steps.length, 5, "Length of Steps increased");
   t.equal(sequencer.images.test.steps[1].options.name, "channel", "Correct Step Inserted");
   t.end();
 });
 
 test('insertSteps({image: {index: index, name: "module", o: options} }) inserts a step', function (t) {
-  sequencer.insertSteps({test: {index:1, name:'channel', o:{} } });
+  sequencer.insertSteps({ test: { index: 1, name: 'channel', o: {} } });
   t.equal(sequencer.images.test.steps.length, 6, "Length of Steps increased");
   t.equal(sequencer.images.test.steps[1].options.name, "channel", "Correct Step Inserted");
   t.end();
 });
 
 test('run() runs the sequencer and returns output to callback', function (t) {
-  sequencer.run('test',function(out){
-    t.equal(typeof(sequencer.images.test.steps[sequencer.images.test.steps.length - 1].output), "object", "Output is Generated");
-    t.equal(out,sequencer.images.test.steps[sequencer.images.test.steps.length - 1].output.src, "Output callback works");
+  sequencer.run('test', function (out) {
+    t.equal(sequencer.images.test.steps.slice(-2)[0].output.getStepOutput(-2).src,
+      sequencer.images.test.steps.slice(-3)[0].output.src, 'offset -2 gives previous step'
```

Comment:
Can we test this by checking the steps names and then comparing to the names of the steps returned by getStep(-1), getStep(0), getStep(1) etc?

Reply:
@jywarren actually this test does the same thing but compares on the output src, Should I add another one which compares step names or delete this one altogether

Reply:
I think we should do both! The `src` may not be a perfect way to assure this, esp. if for some reason it doesn't change. Let's double check!

Reply:
Oh, I get it!! I'll add this test